pkg/services/folder/folderimpl/sqlstore.go

```diff
@@ -268,7 +268,7 @@ func (ss *sqlStore) getParentsMySQL(ctx context.Context, cmd folder.GetParentsQu
 func (ss *sqlStore) GetHeight(ctx context.Context, foldrUID string, orgID int64, parentUID *string) (int, error) {
 	height := -1
 	queue := []string{foldrUID}
-	for len(queue) > 0 {
+	for len(queue) > 0 && height <= folder.MaxNestedFolderDepth {
```

Comment:
even it should not happen, just in case we already have a circular reference, we traverse maximum 8 depth. originally raised by kristin, let's still put it.

Reply:
I think that we better need to handle this on the service level; we want to keep all the logic there.

Reply:
ok, let's keep it but I would suggest to log  also this case then